Code/src/main/java/nl/utwente/group10/haskell/catalog/HaskellClassCatalog.java

```diff
@@ -0,0 +1,132 @@
+package nl.utwente.group10.haskell.catalog;
+
+import nl.utwente.group10.haskell.env.Env;
+import nl.utwente.group10.haskell.exceptions.CatalogException;
+import nl.utwente.group10.haskell.type.TypeClass;
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.validation.SchemaFactory;
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+
+/**
+ * Provides a convenient interface to the XML catalog of available Haskell classes.
+ */
+public class HaskellClassCatalog {
```

Comment:
I have thought about that too, but for know I was looking for a quick-'n-dirty solution to get it working quickly. (In fact, I started working on this but the existing ```HaskellCatalog``` class needed to much adaptation, so I chose to just copy-paste it.) Also, this would break functionality for the UI, so this has to be done very carefully and introduces a lot of work (which I am happy to do, but again, a quick solution for now).

This is another part of the cleanup.